recipes/gegles-spdlog_setup/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.52.0"
+
+
+class SpdlogSetupConan(ConanFile):
+    name = "gegles-spdlog_setup"
+    description = "Setup spdlog via a TOML config file"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/gegles/spdlog_setup"
+    topics = ('spdlog', 'logging', 'header-only', 'toml', 'cpptoml')
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12.0",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
```

Comment:
For easier maintenance in the future, it's better if requirements with `transitive_` traits have a small comment each describing why those are needed. Usually the file/lib is enough, and makes our review life much easier :)

Reply:
Would love to comply with your comments, but I'm not a 100% sure I understand the ask?

Since this is a header-only lib. I thought this was the recommended way to declare its dependencies, am I wrong?

It actively uses all 3 libs directly to do its job...

Reply:
Oh I see!
Note that if you set the package_type to header-library, then these traits are actually inferred to true automatically, so there would be no extra ones to add! See here https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring for more context

Reply:
Thanks! I had not realized that. Now removed in [ef0ccec](https://github.com/conan-io/conan-center-index/pull/17268/commits/ef0ccec7160f34183abbc0c965b3f2ea86b42e2b)